Add comprehensive security enhancements and safety features

abneribeiro opened the pull request:
## Key Features

  ### Security Enhancements
  - **Multi-step confirmation**: 3-tier verification (CONFIRM → random hash → DESTROY)
  - **Comprehensive logging**: All operations logged to `~/.fugo/` with timestamps
  - **Automatic backups**: Creates tar.gz backups before deletion operations
  - **Permission validation**: Validates write access and sudo requirements
  - **Dry-run mode**: Preview operations without execution

  ### Enhanced Detection
  - **Cross-platform support**: Official Go, GVM, Homebrew, package managers
  - **Installation details**: Version, source, size, permissions for each installation
  - **Intelligent discovery**: Detects installations across common paths

  ## Testing
  - 10+ unit tests with benchmark validation
  - Integration tests for detection algorithms
  - Error handling and security function coverage

  ## Breaking Changes
  None - fully backward compatible

  ## Documentation
  - Complete SECURITY.md with usage guide and recovery procedures
  - Troubleshooting and best practices sections

  ## Technical Details
  **New Components:**
  - `GoInstallation` struct for installation metadata
  - `Logger` struct for audit trails
  - Crypto-secure hash generation
  - Backup system with tar compression

  **Files Modified:**
  - Enhanced main detection logic
  - Added security confirmation flows
  - Integrated logging throughout application

abneribeiro committed:
feat: Add comprehensive security enhancements

  - Triple confirmation system with random hash validation
  - Intelligent Go installation detection across platforms
  - Comprehensive logging to ~/.fugo/fugo_TIMESTAMP.log
  - Automatic backup creation before deletion
  - Advanced permission validation and dry-run mode
  - Complete test suite and security documentation

Melkeydev commented on main.go:
```diff
@@ -84,6 +89,102 @@ var criticalPaths = []string{
+		return fmt.Errorf("unable to determine current user: %v", err)
+	}
+
+	if runtime.GOOS != "windows" && currentUser.Uid != "0" {
```
Why are we performing a check on the Uid != 0?

Melkeydev commented on main.go:
```diff
@@ -84,6 +89,102 @@ var criticalPaths = []string{
+
+	var cmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		cmd = exec.Command("tar", "-czf", backupPath, "-C", filepath.Dir(sourcePath), filepath.Base(sourcePath))
+	} else {
+		cmd = exec.Command("tar", "-czf", backupPath, "-C", filepath.Dir(sourcePath), filepath.Base(sourcePath))
```
These cmd look exactly the same

Melkeydev replied:
Would you be able to do that in a follow up commit push so I can resolve the comment

abneribeiro replied:
Yep, just pushed the update!

Melkeydev commented on main.go:
```diff
@@ -380,54 +764,100 @@ func (m model) View() string {
-		s += confirmButtonStyle.Render("ENTER") + " to continue, " + cancelButtonStyle.Render("q") + " to quit\n"
+
+		// Dry run status
+		if m.dryRun {
+			s += highlightStyle.Render("🔍 DRY RUN MODE ENABLED - No files will be deleted") + "\n"
+		} else {
+			s += warningStyle.Render("🔥 LIVE MODE - Files WILL be permanently deleted!") + "\n"
+		}
```
dry run mode is a good feature to include

abneribeiro replied:
Glad you liked it, thanks!

Melkeydev commented on main.go:
```diff
@@ -84,6 +89,102 @@ var criticalPaths = []string{
+		return nil
+	}
+
+	backupName := fmt.Sprintf("go_backup_%s.tar.gz", time.Now().Format("20060102_150405"))
+	backupPath := filepath.Join(backupDir, backupName)
```
The created backups - do we let the users know where the backups are stored via the TUI?

Melkeydev commented on main.go:
```diff
@@ -149,9 +271,164 @@ func (m model) Init() tea.Cmd {
+		return "go version " + strings.TrimSpace(string(data))
+	}
+
+	return "unknown version"
```
I am not a fan of this explicit string return. When we call the `getGoVersion` func and it returns the "unknown version", do we explicitly need to check the string? Since this func doesnt return an error how do we handle the unknown path here?

Melkeydev commented on main.go:
```diff
@@ -149,9 +271,164 @@ func (m model) Init() tea.Cmd {
+	return "unknown version"
+}
+
+func getDirSize(path string) int64 {
+	var size int64
+	filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return nil
+		}
+		if !info.IsDir() {
+			size += info.Size()
+		}
+		return nil
+	})
+	return size
+}
```
nit: is this needed?

abneribeiro replied:
getDirSize() is needed - it's used to display installation sizes in the TUI. The function calculates directory sizes that are 
  shown to users as "💾 Size: X.X MB".
  
![image](https://github.com/user-attachments/assets/7ad39aea-30f4-48e5-ab90-0a2cdb0295f2)

Melkeydev commented on main.go:
```diff
@@ -149,9 +271,164 @@ func (m model) Init() tea.Cmd {
+func getPermissions(path string) string {
+	info, err := os.Stat(path)
+	if err != nil {
+		return "unknown"
```
nit: again with explicit string returns with no error seem like a smell

Melkeydev commented on main.go:
```diff
@@ -346,6 +685,51 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch m.confirmationStep {
+	case 0:
+		if strings.ToUpper(input) == "CONFIRM" {
+			m.confirmationStep = 1
```
nit: confirmation steps should be iota values here but its okay

abneribeiro replied:
I’ve fixed that and switched the confirmation steps to use iota values.

abneribeiro committed:
fix: Remove redundant if/else block in createBackup function

Both Windows and non-Windows branches executed identical tar commands.

abneribeiro committed:
feat: Add additional validations to Go installation test.
Checks the Go installation's version, size, and permissions.
Removes redundant style in the main code.
Improves control structure in the Update method for better clarity.

abneribeiro committed:
feat: Improve Go version and permissions validation

Refactor getGoVersion and getPermissions to return errors. Update tests and add constants for confirmation steps.

Melkeydev reviewed (approved):
LGTM!
